wordpress-mobile/WordPress-iOS
# Remove the .showLinkMailView segue and replace with programmatic nav

Fixes # n/a

This PR removes the segue that sends a user to the "Magic Link sent!" screen. It now navigates the user programmatically.

See also: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/217

## To test:
1. Check out this branch
2. `rake dependencies`
3. Build and run
4. Navigate to and from each of these flows:

#### Flow 1, Login
1. Select the Log in button
2. Select Continue with WordPress.com
3. Enter a valid email address that can use Magic Links, select the Next button
4. Select the Send Link button 
5. Verify you are navigated to the "Magic link sent!" screen
6. Verify you received an emailed magic link
7. Open the magic link
8. Verify the magic link navigates you to the "Continue" screen

#### Flow 2, Signup
1. Select the Sign up for WordPress.com button
2. Select the Sign up with Email button
3. Enter a valid email address that can use Magic Links, select the Next button
4. Select the Send Link button
5. Verify you are navigated to the "Magic link sent!" screen
6. Verify you received an emailed magic link
7. Open the magic link
8. Verify the magic link navigates you to the New Account Creation screen
9. Verify tapping "Continue" navigates you to the "Create a new site" screen

### Do not merge until:
1. The Authenticator PR has been merged: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/217 ✔️
2. A new Authenticator release has been created  ✔️
3. The release is published on Cocoapods trunk ✔️
4. This PR points to the new release  ✔️

PR submission checklist:
